### PR TITLE
Merge dev_postgresql into the master branch

### PR DESCRIPTION
**Pull request by lzjpaul**
Merge dev_postgresql into the master branch

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -559,3 +559,59 @@ PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
...
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+===============================================================================
+SINGA bundles the following under The PostgreSQL License: 
+examples/model_selection/TRAILS-Database-Native-Model-Selection/internal/pg_extension/*
```
Comment (nudles) [inline LICENSE:595]: the same set of files are under MIT and PostgreSQL?
if these are written by SINGA contributor (i.e., not copied from other projects), they should just be released under apache license.

File: examples/model_selection/TRAILS-Database-Native-Model-Selection/internal/pg_extension/sql/pg_extension--0.1.0.sql [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*************************************************************/
+
+/*
+This file is auto generated by pgrx.
```
[the remark this answers is not in the document]
Comment (lzjpaul) [reply examples/model_selection/TRAILS-Database-Native-Model-Selection/internal/pg_extension/sql/pg_extension--0.1.0.sql:23]: Yes, this file can be removed

File: examples/model_selection/TRAILS-Database-Native-Model-Selection/documents/image-20231020174945226.png
[the remark this answers is not in the document]
Comment (lzjpaul) [reply examples/model_selection/TRAILS-Database-Native-Model-Selection/documents/image-20231020174945226.png:1]: yes, will exclude the binary files and state the self-created binary files in the notice file

File: examples/armnet/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    under the License.
+-->
+
+## ARM-Net: Adaptive Relation Modeling Network for Structured Data
+
+![version](https://img.shields.io/badge/version-v3.5-green)
+![python](https://img.shields.io/badge/python-3.8.3-blue)
+![singa](https://img.shields.io/badge/singa-3.1.0-orange)
+
+This folder contains our Singa implementation of [ARM-Net: Adaptive Relation Modeling Network for Structured Data](https://dl.acm.org/doi/10.1145/3448016.3457321).
```
[the remark this answers is not in the document]
Comment (lzjpaul) [reply examples/armnet/README.md:24]: For this release, we will remove this part

File: examples/model_selection/TRAILS-Database-Native-Model-Selection/internal/ml/model_selection/requirement.txt [hunk 4]
```diff
@@ -0,0 +1,54 @@
+aiofiles==23.1.0
+blessed==1.20.0
```
[the remark this answers is not in the document]
Comment (lzjpaul) [reply examples/model_selection/TRAILS-Database-Native-Model-Selection/internal/ml/model_selection/requirement.txt:2]: binary files, config files, and requirement.txt files can not include the license header

File: examples/model_selection/TRAILS-Database-Native-Model-Selection/internal/pg_extension/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*************************************************************/
+
```
[the remark this answers is not in the document]
Comment (lzjpaul) [reply examples/model_selection/TRAILS-Database-Native-Model-Selection/internal/pg_extension/src/lib.rs:21]: these .rs files are written by us

File: LICENSE (hunk 1 above)
Comment (lzjpaul) [reply LICENSE:595]: these source files are written by us, so they should be released under apache license

**Pull request merged**